databricks/koalas/series.py

```diff
@@ -3409,6 +3409,90 @@ def replace(self, to_replace=None, value=None, regex=False) -> 'Series':
 
         return self._with_new_scol(current)
 
+    def update(self, other):
+        """
+        Modify Series in place using non-NA values from passed Series. Aligns on index.
+
+        Parameters
+        ----------
+        other : Series
+
+        Examples
+        --------
+        >>> from databricks.koalas.config import set_option, reset_option
+        >>> set_option("compute.ops_on_diff_frames", True)
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, 5, 6]))
+        >>> s
+        0    4
+        1    5
+        2    6
+        Name: 0, dtype: int64
+
+        >>> s = ks.Series(['a', 'b', 'c'])
+        >>> s.update(ks.Series(['d', 'e'], index=[0, 2]))
+        >>> s
+        0    d
+        1    b
+        2    e
+        Name: 0, dtype: object
+
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, 5, 6, 7, 8]))
+        >>> s
+        0    4
+        1    5
+        2    6
+        Name: 0, dtype: int64
+
+        >>> s = ks.Series([1, 2, 3], index=[10, 11, 12])
+        >>> s
+        10    1
+        11    2
+        12    3
+        Name: 0, dtype: int64
+
+        >>> s.update(ks.Series([4, 5, 6]))
+        >>> s
+        10    1
+        11    2
+        12    3
+        Name: 0, dtype: int64
+
+        >>> s.update(ks.Series([4, 5, 6], index=[11, 12, 13]))
+        >>> s
+        10    1
+        11    4
+        12    5
+        Name: 0, dtype: int64
+
+        If ``other`` contains NaNs the corresponding values are not updated
+        in the original Series.
+
+        >>> s = ks.Series([1, 2, 3])
+        >>> s.update(ks.Series([4, np.nan, 6]))
+        >>> s
+        0    4.0
+        1    2.0
+        2    6.0
+        Name: 0, dtype: float64
+
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, Series):
+            raise ValueError("'other' must be a Series")
+
+        index_scol_name = self._index_map[0][0]
+        combined = combine_frames(self._kdf, other._kdf, how='leftouter')
+        combined_sdf = combined._sdf.sort(index_scol_name)
+        cond = F.when(combined_sdf['__that_0'].isNotNull(), combined_sdf['__that_0']) \
+                .otherwise(combined_sdf['__this_0']) \
+                .alias(self.name)
+        internal = _InternalFrame(
+            sdf=combined_sdf.select(index_scol_name, cond),
+            index_map=self._index_map)
```

Comment:
`column_index` will be lost?

Reply:
sorry i don't get it perfectly 😿 
you mean `s._internal.column_index` will be lost??

Reply:
Yes, now that we use it to manage Series name, so I think we need to keep it.

Reply:
got it, i just checked the `column_index` wouldn't be lost

```python
>>> s1 = ks.Series([1, 2, 3], index=[10, 11, 12], name='ueshin')
>>> s2 = ks.Series([4, 5, 6], index=[11, 12, 13])
>>> s1._internal.column_index
[('ueshin',)]

>>> s1.update(s2)
>>> s1
10    1
12    5
11    4
Name: ueshin, dtype: int64

>>> s1._internal.column_index
[('ueshin',)]
```

Reply:
What if `name=('it', 'holic')`?

Reply:
oops.. it raises `AssertionError`, i'm going to fix it 😭

Reply:
+ the error raised when `combine_frames` , and maybe i think this case seems slightly different issue what you commented above, right?

```python
>>> combined = combine_frames(self.to_frame(), other.to_frame(), how='leftouter')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/utils.py", line 109, in combine_frames
    column_index=column_index, column_index_names=column_index_names))
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/internal.py", line 699, in copy
    column_index=column_index, column_index_names=column_index_names)
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/internal.py", line 444, in __init__
    assert len(set(len(i) for i in column_index)) <= 1, column_index
AssertionError: [('this', 'it', 'holic'), ('that', '0')]
```

Reply:
ah, yeah, that's a bug in `combine_frames`. Let me fix it.

Reply:
Thanks :) then is it okay if i'll fix it after that bug be solved??

Reply:
Yes, please wait for a while. Thanks.

Reply:
yup. please don't need to rush, just let me know when you're finished 😄